lib/blocks.php

```diff
@@ -27,6 +27,39 @@ function gutenberg_get_registered_social_link_blocks() {
 	return $registered_social_link_blocks;
 }
 
+/**
+ * Registers blocks using block manifests discovered in block library.
+ *
+ * @since 6.9.0
+ */
+function gutenberg_register_block_types() {
+	$registry = WP_Block_Type_Registry::get_instance();
+
+	$block_manifests = glob( dirname( dirname( __FILE__ ) ) . '/packages/block-library/src/*/block.json' );
```

Comment:
I don't think we have `block.json` for all core blocks, those which are dynamic don't have this metadata file provided because we still didn't resolve the following issues:
- `supports` is implemented only on the client-side
- translations aren't tackled for fields defined in the JSON file, there is the proposal for the client-side https://github.com/WordPress/gutenberg/pull/16088, but we don't have anything for the server-side

Those aren't blockers for this proposal if we were to use only `attributes` though. So maybe it would be a good idea to move attributes to the `block.json` file to better promote this format.

Reply:
> we still didn't resolve the following issues:

You'll have to forgive me since it's been a while that I've revisited some of those specific details of the JSON manifest. Working through this prototype forced me to consider how we _would_ implement at least some of those supports on the server (`className`, `align`, `anchor` are implemented in this pull request).

For translations, I seem to recall something about how we considered to wrap the translateable fields via `__` et. al., automatically? I'm not sure exactly how we determine the domain in that case.

As a prototype, I'm also fine to start splitting those off into their own individual tasks. It was at least interesting to explore the feasibility of pulling them in and highlighting some of these shortcomings (notably `supports`).

Reply:
> You'll have to forgive me since it's been a while that I've revisited some of those specific details of the JSON manifest. Working through this prototype forced me to consider how we _would_ implement at least some of those supports on the server (`className`, `align`, `anchor` are implemented in this pull request).

Nice, I missed that, sorry about it :(


> For translations, I seem to recall something about how we considered to wrap the translateable fields via `__` et. al., automatically? I'm not sure exactly how we determine the domain in that case.

There needs to be the `textDomain` field declared in the `block.json` file. You can check my prototype for JS side as a reference: https://github.com/WordPress/gutenberg/pull/16088.

I don't think it is a concern though in the context of `attributes`. I just wanted to raise awareness of that. The general agreement was that `attributes` shouldn't be translatable.

> As a prototype, I'm also fine to start splitting those off into their own individual tasks. It was at least interesting to explore the feasibility of pulling them in and highlighting some of these shortcomings (notably `supports`).

Yes, `supports` seems like the only place which can cause issues for the proposed code.